vowpalwabbit/parser.h

```diff
@@ -35,6 +37,46 @@ struct example_initializer
   example* operator()(example* ex);
 };
 
+struct IO_State {
+
+      std::queue<IO_Item> *io_lines = nullptr;
+      std::atomic<bool> done_with_io;
+
+      IO_State(){
+        io_lines = new std::queue<IO_Item>;
+         done_with_io.store(false);
+      }
+
+      IO_State(std::queue<IO_Item> *new_input_lines){
+          //input_lines_copy now points to new_input_lines
+          io_lines = new_input_lines;
+          done_with_io.store(false);
+      }
+
+      IO_State operator=(const IO_State &toCopy){
```

Comment:
Given that you are implementing the copy constructor you should either implement or delete the move constructors.